app.py

```diff
@@ -0,0 +1,178 @@
+                    else:
+                        print ("the option given to create room is invalid")
+                    new_room = [room]
+                    self.room_directory.extend(new_room)
+                else:
+                    return "the room already exists"
+            elif not isinstance(rm_type, str):
+                return "the room option is invalid"
+            elif not isinstance(rm_name, str):
+                return "the room name is invalid"
+
+    def get_available_space(self, room_type):
+        """ filter the room directory to return available rooms for occupancy """
+        if room_type == "o":
+            office_w_space = [room for room in self.room_directory
+                if (len(room.occupants)< room.capacity) and isinstance(room, Office)]
+            return office_w_space
+        elif room_type == "l":
+            living_w_space = [room for room in self.room_directory
+                if (len(room.occupants)< room.capacity) and isinstance(room, LivingSpace)]
+            return living_w_space
+
+    def get_rooms_w_people(self):
+        """ filter room directory to return rooms with occupants"""
+        rooms_w_people = [room for room in self.room_directory
+            if len(room.occupants)>=1]
+        return rooms_w_people
+
+
+    def allocate(self):
+        """ assign people to rooms """
+        available_office = self.get_available_space("o")
+        available_living = self.get_available_space("l")
+        #make a copy of waiting list to allow remove() method to run bug free
+        iteration_list = [guys for guys in self.waiting_list]
+        if len(available_office) == 0 and len(available_living) == 0:
+            print("Can't allocate no Rooms available")
+        elif len(available_office) >= 1:
+            for person in iteration_list:
+                if person.role == "staff":
+                    allocated_office = random.choice(available_office)
+                    allocated_office.occupants.append(person)
+                    self.waiting_list.remove(person)
+                elif person.role == "fellow" and person.wants_living.lower() == "n":
+                    allocated_office = random.choice(available_office)
+                    allocated_office.occupants.append(person)
+                    self.waiting_list.remove(person)
+                elif person.role == "fellow" and person.wants_living.lower() == "y":
+                    if len(available_living) >= 1:
+                        allocated_office = random.choice(available_office)
+                        allocated_living = random.choice(available_living)
+                        allocated_living.occupants.append(person)
+                        allocated_office.occupants.append(person)
+                        self.waiting_list.remove(person)
+                    else:
+                        self.living_waiting_list.append(person)
+                        return "Can't allocate Living Space"
+
```

Comment:
No need to use `if len(available_office) == 0` and then `elif len(available_office) >= 1:` why not just use'
```
if len(avalilable_office) and len(available_living):
    Do your stuff
print("Can't allocate no Rooms available")